### ISSUE TITLE
In the optional "Make a content collection" - post.slug returns undefined

### ISSUE THREAD

**Issue by jimydavis** [withastro/blog-tutorial-demo#35]
in `[...slug].astro` slug is passed as 

```
return posts.map((post) => ({
    params: { slug: post.id },
    props: { post },
  }));
```

It looks like this slug only affects the dynamic route in this folder. When calling post.slug from other places (e.g. blog.astro), it returns undefined.

**Comment by delucis** [comment]
Thanks for the report. We probably need to use `post.id` in those other places. That changed recently in Astro v5.

**Comment by delucis** [comment]
@jimydavis Would you mind sharing where exactly you saw this issue? I searched in the docs but couldn’t find anywhere we show `post.slug`.

**Comment by jimydavis** [comment]
You have to look at the branch content-collection

e.g. https://github.com/withastro/blog-tutorial-demo/blob/content-collections/src/pages/blog.astro#L13

**Comment by delucis** [comment]
Ahhh, thank you! I guess a few things were missed in #33.

**Comment by ArmandPhilippot** [comment]
Yeah, with the content collections we no longer need the extra types added to please Typescript... so using `any` masked the error...
It will be fixed with #36!

**Comment by delucis** [comment]
Fixed in #36!
